src/test/ui/doc-alias-same-name.rs

```diff
@@ -0,0 +1,8 @@
+// compile-flags: -Zdeduplicate-diagnostics=no
+
+#![feature(doc_alias)]
+
+#![crate_type = "lib"]
+
+#[doc(alias = "Foo")] //~ ERROR
```

Comment:
Hmm, I didn't realize this runs in rustc_attr itself. It shouldn't be changed here, but I wonder if it makes sense to run those checks in rustdoc instead?